src/include/duckdb/common/types/value_map.hpp

```diff
@@ -0,0 +1,34 @@
+//===----------------------------------------------------------------------===//
+//                         DuckDB
+//
+// duckdb/parser/expression_map.hpp
+//
+//
+//===----------------------------------------------------------------------===//
+
+#pragma once
+
+#include "duckdb/common/unordered_map.hpp"
+#include "duckdb/common/unordered_set.hpp"
+#include "duckdb/common/types/value.hpp"
+
+namespace duckdb {
+
+struct ValueHashFunction {
+	uint64_t operator()(const Value &value) const {
+		return (uint64_t)value.Hash();
+	}
+};
+
+struct ValueEquality {
+	bool operator()(const Value &a, const Value &b) const {
+		return Value::ValuesAreEqual(a, b);
```

Comment:
You will probably want to use `Values::NotDistinctFrom` here. `ValuesAreEqual` is a bit of a misnomer (and should probably be renamed) - it checks for approximate equivalence in the case of e.g. floating point numbers. This is used for tests and such.